### PR TITLE
`WIP-4091-data` --> `main`

### PR DESCRIPTION
**Pull request by Jooty**
### Client
- Added persistent character & account data.
- Added user authentication.
- Added client session details and validation.
- Added world and zone events.
- Added point-of-interest text.

- Possibly fixed the quit button not working (?)
- Possibly fixed crashed clients not having their object removed from the zone (?)

### Developer
- Rebranded **Imlight**.
- Added `Account` model.
- Added `Character` model.
- Added `CharacterObjectLoader` to load character game objects from their `Character` model.
- Added _RavenDb_ dependencies.
- Added _dragon_, Imlight's database cluster.
- Added _Imlight_, a database containing all of Imlight's world data, events, and zone data. <mark style="background: #FFB8EBA6;">This will soon become embedded</mark>, but remains on a server so that all developers may contribute to it.
- Added _Playerdata_, a database containing all player and account data.
- Added security certificates to _RavenDb_ for individual database access.
- Added `Character` results type (`Result`) to affect a player's stats or behaviors.
- Added `ElementChangeCacheManager` to buffer persistent changes for a class every _x_ minutes.
- Added a few collections for database abstractions:
	- `RavenDatabaseSingleton`
	- `AccountCollection`
	- `CharacterCollection`
	- `ClientKeyCollection`
	- `PlayerDatabase`
	- `WorldDatabase`
	- `WorldDataCollection`
	- `ZoneDataCollection`
- Added a few models for wizard zone and global zone data:
	- `WizardClassData` -- Player base stats and calculations
	- `WizardWorldData` -- Global event data
	- `WizardZoneData`  -- Specific zone event data
	- `WizardZoneEventData` -- Event model
- Added a single source configuration file `Imlight.ini` rather than sourcing from `const` and `static readonly` values. Check #34 for full changelog.
- Added an embedded database option for both databases in case a certificate is not present or the developer has not put in a url.
- Added `OnPreDispose()` to `MessageService` to allow services to allow disposal that may require another service to still be active.
- Added `OnDispose()` to `MessageService` to properly dispose without relying on a message handler.

##### Imlight will now blacklist most objects that are part of events or downright restricted.
- Changed zones to check active events for their objects.
- Changed zones to check for blacklisted adjectives for their objects.
- Changed Wizard City birthday to be between the periods September 2nd and September 12th.
- Changed Christmas event to be between periods December 1st and December 31st.
- Changed Halloween event to be between periods October 1st and October 31st.
- Changed Imlight to be able to run without an active patch server connection.

> [!note] Information
> This branch also brings alongside [[WIP-4092-config]].

## Help Needed
- [x] Check if zone events actually work between their respective time periods.
- [x] Multiplayer test.
- [ ] Change deployment workflow to use proper dragon database certificates.
- [x] Zone transfers must be rebuilt on dragon.

### CODE REVIEW

File: src/Backend/Config/Imlight.ini
Comment (gmrush) [inline src/Backend/Config/Imlight.ini:1]: Excellent addition

**Review by gmrush — APPROVED** [review]
Phenomenal.

**Comment by Jooty** [comment]
It seems somewhere with new zone transfers the POI text broke. I'm also noticing the birthday objects are not active despite the dates being met.

edit: going to leave the global event objects to be manual activation rather than timestamps.

**Comment by Jooty** [comment]
> Change deployment workflow to use proper dragon database certificates.

Moving this last task of the PR. We're reestablishing our branching conventions and the task has moved to that sprint rather than this one. With that gone, this is ready to PR.

**Pull request merged**